cookbook/workflows/README.md

```diff
@@ -0,0 +1,499 @@
+
+- 🔄 **Flexible Execution**: Sequential, parallel, conditional, and loop-based execution
+- 🎯 **Smart Routing**: Dynamic step selection based on content analysis
+- 🔧 **Mixed Components**: Combine agents, teams, and functions seamlessly
+- 💾 **State Management**: Share data across steps with session state
+- 🌊 **Streaming Support**: Real-time processing with streaming capabilities
+- 📝 **Structured Inputs**: Type-safe inputs with Pydantic models
+
+## Core Concepts
+
+### Building Blocks
+
+| Component | Purpose | Example Use Case |
+|-----------|---------|------------------|
+| **Step** | Basic execution unit | Single research task |
+| **Agent** | AI assistant with specific role | Content writer, researcher |
+| **Team** | Coordinated group of agents | Research team with specialists |
+| **Function** | Custom Python logic | Data processing, API calls |
+| **Parallel** | Concurrent execution | Multiple research streams |
+| **Condition** | Conditional execution | Topic-specific processing |
+| **Loop** | Iterative execution | Quality-driven research |
+| **Router** | Dynamic routing | Content-based step selection |
+
+
+## Workflow Patterns
+
+### 1. Sequential Workflows
+
+**When to use**: Linear processes where each step depends on the previous one.
+
+**Example**: Research → Analysis → Content Creation
+
+```python
+from agno.workflow.v2 import Step, Workflow
+
+# Simple sequential workflow
+workflow = Workflow(
+    name="Content Creation Pipeline",
+    steps=[
+        Step(name="Research", agent=researcher),
+        Step(name="Analysis", agent=analyst), 
+        Step(name="Writing", agent=writer),
+    ]
+)
+```
+
+**Key Benefits**:
+- Simple to understand and debug
+- Clear data flow
+- Perfect for linear processes
+
+**See**: [`sequence_of_steps.py`](sync/sequence_of_steps.py)
+
+### 2. Parallel Execution
+
+**When to use**: Independent tasks that can run simultaneously to save time.
+
+**Example**: Multiple research sources, parallel content creation
+
+![Parallel Steps](/cookbook/workflows/assets/parallel_steps.png)
+
+```python
+from agno.workflow.v2 import Parallel, Step, Workflow
+
+workflow = Workflow(
+    name="Parallel Research Pipeline",
+    steps=[
+        Parallel(
+            Step(name="HackerNews Research", agent=hn_researcher),
+            Step(name="Web Research", agent=web_researcher),
+            Step(name="Academic Research", agent=academic_researcher),
+            name="Research Phase"
+        ),
+        Step(name="Synthesis", agent=synthesizer),
+    ]
+)
+```
+
+**Key Benefits**:
+- Faster execution
+- Resource efficiency
+- Independent task handling
+
+**See**: [`parallel_steps_workflow.py`](sync/parallel_steps_workflow.py)
+
+### 3. Conditional Workflows
+
+**When to use**: Different processing paths based on content analysis or business logic.
+
+**Example**: Topic-specific research strategies, content type routing
+
+![Condition Steps](/cookbook/workflows/assets/condition_steps.png)
```

Comment:
This image doesn't correctly show how skip works?

Reply:
added